Mergin/attachment_fields_model.py

```diff
@@ -0,0 +1,43 @@
+from qgis.PyQt.QtGui import QStandardItemModel, QStandardItem
+from qgis.PyQt.QtCore import Qt, QSize
+from qgis.core import QgsProject, QgsMapLayer, QgsSymbolLayerUtils, QgsIconUtils
+
+
+class AttachmentFieldsModel(QStandardItemModel):
+    LAYER_ID = Qt.UserRole + 1
+    FIELD_NAME = Qt.UserRole + 2
+    EXPRESSION = Qt.UserRole + 3
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self.setHorizontalHeaderLabels(["Layer", "Field"])
+
+        parent_item = self.invisibleRootItem()
+
+        layers = QgsProject.instance().mapLayers()
+        for layer_id, layer in layers.items():
+            if layer.type() != QgsMapLayer.VectorLayer:
+                continue
+
+            for field in layer.fields():
+                widget_setup = field.editorWidgetSetup()
+                if widget_setup.type() != "ExternalResource":
+                    continue
+
+                if not widget_setup.config().get("DocumentViewer", 1):
+                    continue
```

Comment:
We need to remove the check for `DocumentViewer` here as well as we removed it in mobile